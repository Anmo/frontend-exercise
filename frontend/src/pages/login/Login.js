import React, { useRef, useCallback } from 'react';
import PropTypes from 'prop-types';
import {
    getUser,
} from '../../shared/utils/api';
import withSession from '../../shared/contexts/session/withSession';

import styles from './Login.module.css';

const Login = ({updateUserState}) => {
    const usernameRef = useRef();
    const handleLogin = useCallback(async (event) => {
        event.preventDefault();
        const username = usernameRef.current.value;

        if(username === '') return;

        try {
            const response = await getUser(username);
            const user = response.data.user;
            updateUserState(user);
        } catch(e) {
            console.error(e);
        }
    }, [updateUserState]);

    return (
        <div className={ styles.container }>
            <h1 variant="h3" component="h1" margin="normal" className={ styles.title }>
                Login
            </h1>
            <form onSubmit={ handleLogin }>
                <label className={ styles.label }>Enter your name</label>
                <input
                    name="username"
                    label="Enter your name"
                    ref={ usernameRef }
                    className={ styles.field } />
                <button
                    type="submit"
                    className={ styles.button }
                    disabled={ usernameRef.current && usernameRef.current.value === ''}>
                    Sign in
                </button>
            </form>
        </div>
    );
}

Login.propTypes = {
    updateUserState: PropTypes.func,
}

export default withSession(Login);