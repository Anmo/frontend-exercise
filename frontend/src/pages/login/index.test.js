import { render } from '@testing-library/react';
import { useLocation } from 'react-router-dom';

import Login from './index';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn()
}))

test('renders the normal Login', () => {
  useLocation.mockImplementation(() => ({}));

  const { container } = render(<Login />);

  expect(container.firstChild).toMatchSnapshot()
});
