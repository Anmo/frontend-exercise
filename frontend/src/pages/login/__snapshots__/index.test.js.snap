// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the normal Login 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    class="container"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
        data-shrink="true"
        for="username"
        id="username-label"
      >
        User Name
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          autocomplete="username"
          class="MuiInputBase-input MuiOutlinedInput-input"
          data-testid="username"
          id="username"
          name="username"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
          >
            <span>
              User Name
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Sign In
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</main>
`;
